axelor-account/src/main/java/com/axelor/apps/account/service/payment/invoice/payment/InvoicePaymentValidateServiceImpl.java

```diff
@@ -467,6 +467,7 @@ public Move fillMove(
         move.addMoveLineListItem(financialDiscountVatMoveLine);
       }
     }
+    move.setInvoice(invoice);
```

Comment:
We only set the invoice field on a Move if it's a Move generated from an Invoice, here is not the case, it's a payment Move.